### PR TITLE
Validate datasets with in-house validator

### PR DESCRIPTION
**Pull request by maukoquiroga**
Relates to #294 

This pull request provides a concurrent and fault-tolerant replacement of the current dataset validation. Validation-by-task interface is unchanged.

### CODE REVIEW

File: lib/mix/tasks/transport/validate_data.ex [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
 
   use Mix.Task
-  alias Transport.ReusableData.Dataset
+  alias Transport.ReusableData
   alias Transport.DataValidation
   require Logger
 
-  @pool DBConnection.Poolboy
+  @concurrency 5
+  @timeout :infinity
```
Comment (Tristramg) [inline lib/mix/tasks/transport/validate_data.ex]: The biggest file we can have, is the GTFS of ÎdF. It’s currently 60Mb, but might grow up to 100Mb at certain moments of the year.
The validation takes 15seconds, so I guess that 1 minute would be a reasonable timeout.

Concerning the concurrency, right now the validator is able to process everything. So a 5 parallel should be fairly conservative

File: lib/transport/data_validation/data_validation.ex [hunk 2]
```diff
@@ -3,98 +3,45 @@ defmodule Transport.DataValidation do
...
   """
-  @spec find_project(map()) :: {:ok, Project.t} | {:error, any()}
-  def find_project(%{} = params) do
-    params
-    |> FindProject.new
-    |> Project.execute
+  @spec find_dataset(map) :: {:ok, Dataset.t()} | {:error, any}
+  def find_dataset(%{} = attrs) do
+    attrs
+    |> FindDataset.new()
```
[the remark this answers is not in the document]
Comment (maukoquiroga) [reply lib/transport/data_validation/data_validation.ex]: They were added automatically by `mix format`

File: lib/mix/tasks/transport/validate_data.ex (hunk 1 above)
Comment (maukoquiroga) [reply lib/mix/tasks/transport/validate_data.ex]: OK

File: lib/transport/data_validation/aggregates/dataset_validation.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  defstruct [
+    :issue_type,
+    :object_id,
+    :object_name,
+    :related_object_id,
+    :severity
+  ]
+
+  @type t :: %__MODULE__{
+          issue_type: String.t(),
```
[the remark this answers is not in the document]
Comment (maukoquiroga) [reply lib/transport/data_validation/aggregates/dataset_validation.ex]: An over-engineered answer then: from the [Erlang documentation](http://erlang.org/doc/man/erl.html#+t), by default, the maximum numbers of atoms the virtual machine can handle is 1,048,576, and they're not garbage collected. If a malicious agent somehow gains access to the validator, and starts sending random strings, we'd quite easily under a DoS situation.

**Pull request merged**